warn/warn.go

```diff
@@ -184,8 +212,20 @@ func FileWarnings(f *build.File, pkg string, enabledWarnings []string, fix bool)
 
 	for _, warn := range warnings {
 		if fct, ok := FileWarningMap[warn]; ok {
+			for _, w := range fct(f) {
```

Comment:
for if for if if for ?  Too much nesting.

Can you refactor?

Reply:
Simplified by creating two subfunctions (`runFileWarningsFunction` and `runRuleWarningsFunction`). The legacy code for running the functions with the old interface will be removed soon(ish) anyway, but I'm going to implement replacements on `attr-cfg` first (if that works correctly and I don't need to alter the interface anymore I'll convert all other functions and replacements will become available for them automatically).